app-creator/client/src/utils/helpers.ts

```diff
@@ -35,14 +49,57 @@ export const noop: VoidFunction = () => {};
 export const chips = [
   {
     label: 'All',
-    device: DeviceType.all,
+    device: DeviceType.ALL,
   },
   {
     label: 'Web',
-    device: DeviceType.desktop,
+    device: DeviceType.DESKTOP,
   },
   {
     label: 'Mobile',
-    device: DeviceType.mobile,
+    device: DeviceType.MOBILE,
   },
 ];
+
+/*
+ * Generates random ids with length 32.
+ */
+export function generateRandomId() {
+  return [...Array(32)]
+    .map((_) => (~~(Math.random() * 36)).toString(36))
+    .join('');
+}
+
+/**
+ * Returns a deep clone of template without widgetRefs.
+ */
+export function deepCloneTemplate(
+  template: AppCreatorStore['template'],
+  skipRefs: boolean = true
+): AppCreatorStore['template'] {
+  const clone: AppCreatorStore['template'] = {};
+  for (const key in template) {
+    /**
+     * Widget refs are only needed in the context of the app creator. Once we serialize the data,
+     * we no longer need to keep the refs. As a result, we skip over them when we are producing the
+     * output string.
+     */
+    if (skipRefs && key === WIDGET_REF) {
+      continue;
+    }
+
+    if (
+      typeof template[key] === 'object' &&
+      !Array.isArray(template[key]) &&
```

Comment:
I'll consolidate a few comments in one here:
- You can remove this condition if you move the `Array.isArray` condition up top.
- Do we not care if `key === WIDGET_REF` for the array case?
- Also, we don't apply `deepCloneTemplate` to the elements of the array (of widgets?), so they will not be cloned. Is that the expected behavior? If so, leave a comment about it and/or write a test for it.

Reply:
- The key `WIDGET_REF` only stores a reference to an HTMLElement so there is no need to check it for the array case. 
- Our widgets are stored in an object so the only arrays we have contain primitive data types and thus cloning the array gives us new references. Nonetheless, we could call deepCloneTemplate again on each child.